eth_abi/codec_packed.py

```diff
@@ -0,0 +1,41 @@
+from typing import (
+    Any,
+    Iterable,
+    Tuple,
+)
+import warnings
+
+from eth_typing.abi import (
+    Decodable,
+    TypeStr,
+)
+
+from eth_abi.codec import (
+    ABICodec,
+)
+from eth_abi.registry import (
+    ABIRegistry,
+    registry_packed as default_packed_registry,
+)
+
+warnings.warn(
+    "Packed mode encoding is an experimental feature.  Please report any "
+    "problems at https://github.com/ethereum/eth-abi/issues."
+)
+
+
+class ABICodecPacked(ABICodec):
```

Comment:
Looks like the inheritance actually works the other way, with some methods only implemented on the non-packed codec.

Reply:
Yeah, it's weird. I feel like the assumption is that any subclass of the `ABIEncoder` would be expected to implement all five methods, but the packed implementation is experimental and isn't decodable. Not sure if it makes sense to have a completely different class for that one. Was trying to keep in line with the ABICodec. Thoughts? I could see both ways.

Reply:
Another alternative is to make a third base codec class that implements the encoding methods. The packed codec would be like:

```py
class ABICodecPacked(BaseABICodecEncoding):
    pass
```

The full codec would also subclass the Base codec and implement the decoding.

Reply:
@carver I considered that as well, but would we want to make a `BaseABICoderDecoding` as well?

Reply:
I think it's fine to just implement that in `ABICodec` for now.

Reply:
Updated here: https://github.com/ethereum/eth-abi/pull/109/commits/abceb187b537e3b754356feb3bbfba6a7989684a

Reply:
Added null check for the registry provided to the base encoder: f939865

Reply:
and a test for the null case ^ - d63273c

Reply:
Ended up naming it `BaseABICodecEncoder`. Let me know if you think that doesn't make sense since we have separate `*Encoder` classes. If so, I can make it `BaseABICodecEncoding` like you suggested. Just thought the wording of that was kind of odd. 🤔

Reply:
Yup, seems fine to me. Could also be `BaseABIEncoder` since it's only responsible for the encoding side of the codec. Whatever you like, since it's not intended to be part of the public API. :)